src/concrete/ml/sklearn/linear_model.py

```diff
@@ -103,6 +118,959 @@ def load_dict(cls, metadata: Dict):
+        # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/3373
+        params = super().get_params(deep=deep)  # type: ignore[misc]
+
+        # Remove the parameters added by Concrete ML
+        params.pop("n_bits", None)
+        params.pop("n_bits_training", None)
+        params.pop("rounding_training", None)
+        params.pop("fit_encrypted", None)
+        params.pop("parameters_range", None)
+        params.pop("batch_size", None)
+        params.pop("learning_rate_value", None)
+
+        return params
+
+    def _get_training_quantized_module(
+        self,
+        x_min: numpy.ndarray,
+        x_max: numpy.ndarray,
+    ) -> QuantizedModule:
+        """Get the quantized module for FHE training.
+
+        This method builds the quantized module and fhe-circuit needed to train the model in FHE.
+
+        Args:
+            x_min (numpy.ndarray): The minimum value to consider for each feature over the samples.
+            x_max (numpy.ndarray): The maximum value to consider for each feature over the samples.
+
+        Returns:
+            (QuantizedModule): The quantized module containing the FHE circuit for training.
+
+        """
+        # Mypy
+        assert self.parameters_range is not None
+
+        # Compile and return the training quantized module
+        # 54 = 2 classes * 3 values for x * 2 values for the weights * 2 values for the bias
+        # Number of combination of extreme values
+        combinations = list(
+            itertools.product(
+                [1.0, 0.0],  # Labels
+                [x_min, x_max, numpy.zeros(x_min.shape)],  # Data-range
+                [self.parameters_range[0], self.parameters_range[1]],  # Weights
+                [self.parameters_range[0], self.parameters_range[1]],  # Bias
+            )
+        )
+        compile_size = len(combinations)
+
+        n_targets = 1
+
+        # Generate the input values to consider for compilation
+        x_compile_set = numpy.empty((compile_size, self.batch_size, x_min.shape[0]))
+
+        # Generate the target values to consider for compilation
+        # Update this once we support multi-class
+        # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/4182
+        y_compile_set = numpy.empty(
+            (
+                compile_size,
+                self.batch_size,
+                n_targets,
+            )
+        )
+
+        # Generate the weight values to consider for compilation
+        weights_compile_set = numpy.empty((compile_size, x_min.shape[0], n_targets))
+
+        # Generate the bias values to consider for compilation
+        bias_compile_set = numpy.empty((compile_size, 1, n_targets))
+
+        compile_set = (x_compile_set, y_compile_set, weights_compile_set, bias_compile_set)
+
+        # Bound values are hard-coded in order to make sure that the circuit never overflows
+        for index, (label, x_value, coef_value, bias_value) in enumerate(combinations):
```

Comment:
I feel like this loop just creates a new `compile_set` from `combinations`, unless I'm missing something, I believe `compile_set`'s above definition (l349) is useless no ?

Reply:
everything is empty at first and the loop populates it, I could define it afterwards but that's not really different to what's being done here

Reply:
ah yes ok sorry I get it now, you go from a list of n tuples of 4 elements to a tuple of 4 arrays of shape (n, ...)

I feel like there might be a simpler way (and easier to read) of doing so but I don't see it for now so looks good to me